dmd 2.097.2 build broken on 22.05 (x86_64)

### Describe the bug


### Steps To Reproduce
Steps to reproduce the behavior:
1. update to latest 22.05
2. nix-shell -p dmd


### Expected behavior
Build should work, or be marked as broken.

### Screenshots / Logs

```
(...)
make[1]: Leaving directory '/build/druntime/test/lifetime'
2 modules passed unittests
2 modules passed unittests
make[1]: Leaving directory '/build/druntime/test/gc'
error: builder for '/nix/store/hyj1j2pb9im7akmnsq68wz3lqf5bhcnj-dmd-2.097.2.drv' failed with exit code 2;
       last 10 log lines:
       >     p = 0x7ffff7725000, size = 10, parent = (nil)
       >     p = 0x7ffff7bbe000, size = 32768, parent = (nil)
       >     p = 0x7ffff7bc7000, size = 4113, parent = (nil)
       > 2 modules passed unittests
       > 2 modules passed unittests
       > rm generated/linux/release/64/large_aggregate_destroy_21097
       > make[1]: Leaving directory '/build/druntime/test/lifetime'
       > 2 modules passed unittests
       > 2 modules passed unittests
       > make[1]: Leaving directory '/build/druntime/test/gc'
       For full logs, run 'nix log /nix/store/hyj1j2pb9im7akmnsq68wz3lqf5bhcnj-dmd-2.097.2.drv'.

```


### Notify maintainers
@ThomasMader 
@lionello 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.43, NixOS, 22.05 (Quokka), 22.05.751.8b66e3f2ebc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Here is the relevant failing test and its error:

```console
Testing rt_trap_exceptions_drt_gdb
gdb -n -ex 'set confirm off' -ex run -ex 'bt full' -ex q --args generated/linux/release/64/rt_trap_exceptions_drt --DRT-trapExceptions=0 \
        > ./generated/linux/release/64/rt_trap_exceptions_drt_gdb.output 2>&1 || true
cat ./generated/linux/release/64/rt_trap_exceptions_drt_gdb.output
GNU gdb (GDB) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from generated/linux/release/64/rt_trap_exceptions_drt...
Starting program: /build/druntime/test/exceptions/generated/linux/release/64/rt_trap_exceptions_drt --DRT-trapExceptions=0
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libthread_db.so.1".
src/rt/dwarfeh.d:330: uncaught exception reached top of stack
This might happen if you're missing a top level catch in your fiber or signal handler
object.Exception@src/rt_trap_exceptions_drt.d(4): foo
----------------
src/rt_trap_exceptions_drt.d:4 void rt_trap_exceptions_drt.test() [0x4376f8]
src/rt_trap_exceptions_drt.d:9 _Dmain [0x437708]

Program received signal SIGABRT, Aborted.
0x00007ffff7d48c1f in __pthread_kill_implementation () from /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libc.so.6
#0  0x00007ffff7d48c1f in __pthread_kill_implementation ()
   from /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libc.so.6
No symbol table info available.
#1  0x00007ffff7cfe042 in raise ()
   from /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libc.so.6
No symbol table info available.
#2  0x00007ffff7ce949c in abort ()
   from /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libc.so.6
No symbol table info available.
#3  0x0000000000437fdf in _d_throwdwarf ()
No symbol table info available.
#4  0x00000000004376f9 in _D22rt_trap_exceptions_drt4testFZv ()
    at ../../src/object.d:2478
        innerLocal = -20384
#5  0x0000000000437709 in _Dmain (args=...) at src/rt_trap_exceptions_drt.d:9
        myLocal = 0x437d7b <rt.dmain2._d_run_main2(char[][], ulong, extern(C) int(char[][]) function).runAll().__lambda2()+31> "H\213M\370\211\201P\377\377\377H\213\345]\303"
grep "in D main (args=...) at .*rt_trap_exceptions_drt.d:9" > /dev/null < ./generated/linux/release/64/rt_trap_exceptions_drt_gdb.output
make[1]: *** [Makefile:87: generated/linux/release/64/rt_trap_exceptions_drt_gdb.done] Error 1
rm generated/linux/release/64/future_message generated/linux/release/64/stderr_msg generated/linux/release/64/catch_in_finally generated/linux/release/64/unknown_gc generated/linux/release/64/invalid_memory_operation generated/linux/release/64/static_dtor
make[1]: Leaving directory '/build/druntime/test/exceptions'
make: *** [posix.mak:451: test/exceptions/.run] Error 2
error: builder for '/nix/store/6qka505jsayf4zbz78a424rmhynvd9wd-dmd-2.097.2.drv' failed with exit code 2;
```

Looks like the gdb output has changed slightly so this grep doesn't work anymore.

--- a/pkgs/development/compilers/dmd/default.nix
+++ b/pkgs/development/compilers/dmd/default.nix
@@ -109,6 +109,10 @@ stdenv.mkDerivation rec {
     rm dmd/test/runnable/gdb15729.sh
     rm dmd/test/runnable/gdb4149.d
     rm dmd/test/runnable/gdb4181.d
+
+    # Grep'd string changed with gdb 12
+    substituteInPlace druntime/test/exceptions/Makefile \
+      --replace 'in D main (' 'in _Dmain ('
   ''
 
   + lib.optionalString stdenv.isLinux ''

